src/charm.py

```diff
@@ -19,6 +24,181 @@ class KratosCharm(CharmBase):
+        # Get current layer
+        current_layer = self._container.get_plan()
+        # Create a new config layer
+        new_layer = self._pebble_layer
+
+        if current_layer.services != new_layer.services:
+            self.unit.status = MaintenanceStatus("Applying new pebble layer")
+            self._container.add_layer(self._container_name, new_layer, combine=True)
+            logger.info("Pebble plan updated with new configuration, replanning")
+            self._container.replan()
+
+        # Compare changes in kratos config
+        try:
+            current_config = self._container.pull(self._config_file_path).read()
+        except PathError:
+            self._container.push(self._config_file_path, self._config, make_dirs=True)
+            logger.info("Updated kratos config")
+        else:
+            if current_config != self._config:
+                self._container.push(self._config_file_path, self._config, make_dirs=True)
+                logger.info("Updated kratos config")
+
+        self._container.restart(self._container_name)
+
+    def _get_database_relation_info(self) -> dict:
+        """Get database info from relation data bag."""
+        relation_id = self.database.relations[0].id
+        relation_data = self.database.fetch_relation_data()[relation_id]
+
+        return {
+            "username": relation_data["username"],
+            "password": relation_data["password"],
+            "endpoints": relation_data["endpoints"],
+        }
+
+    def _set_default_identity_schema(self) -> None:
+        """Push identity schema into kratos container."""
+        try:
+            with open("src/identity.default.schema.json", encoding="utf-8") as schema_file:
+                schema = schema_file.read()
+                self._container.push(self._identity_schema_file_path, schema, make_dirs=True)
+            logger.info("Pushed configs to kratos container")
+        except (ProtocolError, PathError) as e:
+            logger.error(str(e))
+            self.unit.status = BlockedStatus(str(e))
+
+    def _update_container(self, event) -> None:
+        """Update configs, pebble layer and run database migration."""
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Kratos container. Deferring pebble ready event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Kratos container")
+            return
+
+        if not self.model.relations["pg-database"]:
+            event.defer()
+            logger.error("Missing required relation with postgresql")
+            self.model.unit.status = BlockedStatus("Missing required relation with postgresql")
+            return
+
+        if not self.database.is_database_created():
+            event.defer()
+            logger.info("Missing database details. Deferring pebble ready event.")
+            self.unit.status = BlockedStatus("Waiting for database creation")
```

Comment:
How about `WaitingStatus` instead? `BlockedStatus` suggests that a manual intervention is required.

Reply:
Good catch, will update that